packages/odyssey-react/src/components/Field/index.tsx

```diff
@@ -0,0 +1,159 @@
+ */
+
+import type { FunctionComponent, ReactElement, ReactText } from "react";
+import ScreenReaderText from "../ScreenReaderText";
+
+import styles from "./Field.module.scss";
+
+export interface SharedFieldTypes {
+  /**
+   * the form field label
+   */
+  label: string;
+
+  /**
+   * Text to display when the field is optional, i.e. required prop is false
+   */
+  optionalLabel?: string;
+
+  /**
+   * the form field error
+   */
+  error?: string;
+
+  /**
+   * the form field hint
+   */
+  hint?: string;
+
+  /**
+   * The underlying input element required attribute
+   * @default true
+   */
+  required?: boolean;
+}
+
+export interface Props extends SharedFieldTypes {
+  /**
+   * Input to be rendered within the Field
+   */
+  children: ReactElement | ReactElement[];
+
+  /**
+   * The underlying input element id attribute. Automatically generated if not provided
+   */
+  inputId: string;
+
+  /**
+   * The underlying parent semantic HTML element.
+   * @default div
+   */
+  as?: "div" | "fieldset";
+}
+
+interface PropsLabel {
+  inputId: string;
+  optionalLabel?: string;
+  required: boolean;
+  children: ReactText;
+  as?: "label" | "legend";
+}
+interface PropsHint {
+  id: string;
+  children: ReactText;
+}
+interface PropsError {
+  id: string;
+  children: ReactText;
+  errorPrefix?: string;
+}
+
+export type StaticComponents = {
+  Label: typeof Label;
+  Hint: typeof Hint;
+  Error: typeof Error;
+};
+
+const Field: FunctionComponent<Props> & StaticComponents = (props) => {
+  const {
+    error,
+    hint,
+    inputId,
+    label,
+    optionalLabel,
+    required = true,
+    children,
+    as = "div",
+  } = props;
+
+  const Tag = as;
+  const TagLabel = as === "fieldset" ? "legend" : "label";
+
+  return (
+    <Tag className={styles.root}>
+      <Field.Label
+        inputId={inputId}
+        required={required}
+        optionalLabel={optionalLabel}
+        as={TagLabel}
+      >
+        {label}
+      </Field.Label>
+      {hint && <Field.Hint id={inputId}>{hint}</Field.Hint>}
+      {children}
+      {error && <Field.Error id={inputId}>{error}</Field.Error>}
+    </Tag>
+  );
+};
+
+const Label = (props: PropsLabel) => {
+  const {
+    inputId,
+    optionalLabel = "Optional",
+    required,
+    children,
+    as = "label",
+  } = props;
+
+  const Tag = as;
+
+  return (
+    <Tag
+      // We'll also need the ability to hide or reposition this label for UI like Search
+      className={styles.label}
+      htmlFor={inputId}
+    >
+      {children}
+      {!required && optionalLabel && (
+        <span className={styles.optionalLabel} children={optionalLabel} />
+      )}
+    </Tag>
+  );
+};
+
+const Hint = ({ id, children }: PropsHint) => (
+  <p className={styles.hint} id={`${id}-hint`} children={children} />
+);
+
+const Error = ({ id, children, errorPrefix = "Error:" }: PropsError) => (
+  <p className={styles.error} id={`${id}-error`}>
+    <ScreenReaderText>{errorPrefix}</ScreenReaderText>
```

Comment:
Fixed!

Reply:
We still have a hardcoded english string on line 158 above! We can't apply english strings even as defaults

Reply:
Hm. This is how we've been handling `optionalLabel` prior to this PR, which is why we used the same approach here.

Double prefixing is a legit concern here, but docs/guidance feel like a better solve for that problem rather than allowing a case where no prefix exists by default. 

Ex: If someone is reviewing UI, "Error: Error: This is a required field" is clearly a bug, but "This is a required field." is not.

@arnoldsandoval, any thoughts here?

Reply:
My concern with allowing ReactNode is that it leaves room for a developer to forget the error prefix, which then creates an accessibility usability problem.

If hardcoded English strings are an issue, then we need to solve for `optionalLabel` as well, which as shred pointed out is implemented in the same fashion. I'm not sure that providing an opinionated default (in English) should be characterized as an i18n bug as we provide developers a way to update the error prefix should their products require it.